Blog.API/Blog.Infrastructure/MediatR/Handlers/Categories/UpdateCategoryHandler.cs

```diff
@@ -1,28 +1,37 @@
-﻿using System.Threading;
-using System.Threading.Tasks;
-using Blog.Core.Entities;
+﻿using Blog.Core.Contracts.Controllers.Categories;
 using Blog.Core.MediatR.Commands.Categories;
 using Blog.Infrastructure.Services.Interfaces;
-using MapsterMapper;
 using MediatR;
+using Microsoft.Extensions.Logging;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace Blog.Infrastructure.MediatR.Handlers.Categories;
 
-public sealed class UpdateCategoryHandler : IRequestHandler<UpdateCategoryCommand, Category?>
+public sealed class UpdateCategoryHandler : IRequestHandler<UpdateCategoryCommand, CategoryResponse?>
 {
     private readonly ICategoryService _categoryService;
-    private readonly IMapper _mapper;
+    private readonly ILogger _logger;
 
-    public UpdateCategoryHandler(ICategoryService categoryService, IMapper mapper)
+    public UpdateCategoryHandler(ICategoryService categoryService,
+        ILogger<UpdateCategoryHandler> logger)
     {
         _categoryService = categoryService;
-        _mapper = mapper;
+        _logger = logger;
     }
 
-    public async Task<Category?> Handle(UpdateCategoryCommand request, CancellationToken cancellationToken)
+    public async Task<CategoryResponse?> Handle(UpdateCategoryCommand request, CancellationToken cancellationToken)
     {
-        var category = _mapper.Map<Category>(request.Category);
-        var responseCategory = await _categoryService.UpdateCategory(category);
+        CategoryResponse? responseCategory = await _categoryService.UpdateCategory(request.Category);
+
+        if (responseCategory is null)
+        {
+            _logger.LogWarning("Category wasn't updated");
```

Comment:
This does not give us enough information. This should be something like this:
```suggestion
            _logger.LogWarning("Category wasn't updated {FailedUpdateCategoryId}", request.Category.CategoryId);
```